fe/fe-core/src/main/antlr4/org/apache/doris/nereids/PLParser.g4

```diff
@@ -324,11 +325,15 @@ package_body_item :
       declare_stmt_item
     | create_function_stmt
     | create_procedure_stmt
+    | drop_procedure_stmt
     ;
 
 create_procedure_stmt :
       (ALTER | CREATE (OR REPLACE)? | REPLACE) (PROCEDURE | PROC) multipartIdentifier create_routine_params? create_routine_options? (AS | IS)? declare_block_inplace? label_stmt? procedure_block (ident_pl SEMICOLON)?
     ;
+drop_procedure_stmt:
+      (DROP  (PROCEDURE | PROC) multipartIdentifier create_routine_params? create_routine_options?)
```

Comment:
why syntax not same with drop proc in DorisParser.g4?

Reply:
from stored procedure can be created using overloading with different parameter. so drop procedure kept similar to stored create procedure for plsql.

Reply:
`create_routine_params` and `create_routine_options` are not required,
refer to /hive/hplsql/src/main/antlr4/org/apache/hive/hplsql/Hplsql.g4,
![image](https://github.com/apache/doris/assets/13197424/d47b681a-e91b-4084-9893-1f691b1b14ab)

we keep the same grammar as hplsql, such as:
`DROP PROCEDURE (IF EXISTS)? name=multipartIdentifier`
or
`DROP (PROCEDURE | PROC) (IF EXISTS)? name=multipartIdentifier`